app/helpers/shop_helper.rb

```diff
@@ -20,27 +20,24 @@ def user_is_related_to_distributor?
     )
   end
 
-  def base_shop_tabs(column_sizes)
+  def shop_tabs
     [
-      { name: 'about', cols: column_sizes[0],
-        title: t(:shopping_tabs_about, distributor: current_distributor.name) },
-      { name: 'producers', cols: column_sizes[1],
-        title: t(:label_producers) },
-      { name: 'contact', cols: column_sizes[2],
-        title: t(:shopping_tabs_contact) }
+      { name: 'home', title: t(:shopping_tabs_home), show: show_home_tab? },
+      { name: 'shop', title: t(:shopping_tabs_shop), show: !require_customer? },
+      { name: 'about', title: t(:shopping_tabs_about), show: true },
+      { name: 'producers', title: t(:label_producers), show: true },
+      { name: 'contact', title: t(:shopping_tabs_contact), show: true },
+      { name: 'groups', title: t(:label_groups), show: current_distributor.groups.any? },
     ]
```

Comment:
You could filter the tabs here and save the `if tab[:show]` in the view:
```rb
[
...
].select { |tab| tab[:show] }
```

It would also make `first_visible_tab` obsolete.